src/ILCompiler.MetadataTransform/src/ILCompiler/Metadata/ExplicitScopeAssemblyPolicyMixin.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Cts = Internal.TypeSystem;
+using Ecma = System.Reflection.Metadata;
+
+namespace ILCompiler.Metadata
+{
+    /// <summary>
+    /// Mixin helper class to allow easy support for ExplicitScopeAttribute for metadata transformation
+    /// 
+    /// ExplicitScopeAttribute is used to relocate where a given type appears in metadata from one 
+    /// metadata assembly to another. It must not be used to relocate into an assembly which otherwise 
+    /// exists within the compilation operation (Current implementation is not reliable in those circumstances.)
```

Comment:
> Current implementation is not reliable in those circumstances

We could guard this with an assert in MetadataTransform.cs (probably near the end of the `Run` method).